lib/dry/schema/processor.rb

```diff
@@ -177,6 +178,20 @@ def rule_applier
         @rule_applier ||= steps.last
       end
       alias_method :to_rule, :rule_applier
+
+      # Check if there are filter rules
+      #
+      # @api private
+      def filter_rules?
+        steps.select { |s| s.is_a?(RuleApplier) }.size > 1
```

Comment:
hm, `steps.any?`?

Reply:
@flash-gordon I know it's weird but it's actually correct - if there are more than two rule appliers, it means we've got a filter schema and the final schema